src/lib/services/feature-type-service.ts

```diff
@@ -27,23 +34,33 @@ export default class FeatureTypeService {
     async updateLifetime(
         id: string,
         newLifetimeDays: number | null,
+        user: IUser,
     ): Promise<IFeatureType> {
         // because our OpenAPI library does type coercion, any `null` values you
         // pass in get converted to `0`.
         const translatedLifetime =
             newLifetimeDays === 0 ? null : newLifetimeDays;
 
+        const featureType = await this.featureTypeStore.get(id);
+
         const result = await this.featureTypeStore.updateLifetime(
             id,
             translatedLifetime,
         );
 
-        if (!result) {
+        if (!featureType || !result) {
             throw new NotFoundError(
                 `The feature type you tried to update ("${id}") does not exist.`,
             );
         }
```

Comment:
If this is the only error case here, would it make sense to move it to before trying to update the lifetime and only check whether it exists? 


```suggestion
        if (!featureType) {
            throw new NotFoundError(
                `The feature type you tried to update ("${id}") does not exist.`,
            );
        }
```

Reply:
tried that originally but TS led me towards my current solution. You can try for yourself. If you add your line and move it before update the return statement will not compile